drivers/tmp00x/tmp00x.c

```diff
@@ -0,0 +1,298 @@
+        return -TMP00X_ERROR_BUS;
+    }
+    i2c_release(BUS);
+
+    return TMP00X_OK;
+}
+
+int tmp00x_reset(const tmp00x_t *dev)
+{
+    uint8_t reg[2];
+    uint16_t tmp = TMP00X_CONFIG_RST;
+    reg[0] = (tmp >> 8);
+    reg[1] = tmp;
+
+    /* Acquire exclusive access to the bus. */
+    i2c_acquire(BUS);
+    if (i2c_write_regs(BUS, ADDR, TMP00X_REGS_CONFIG, reg, 2, 0) < 0) {
+        i2c_release(BUS);
+        return -TMP00X_ERROR_BUS;
+    }
+    i2c_release(BUS);
+    return TMP00X_OK;
+}
+
+int tmp00x_set_active(const tmp00x_t *dev)
+{
+    uint8_t reg[2];
+
+    i2c_acquire(BUS);
+    if (i2c_read_regs(BUS, ADDR, TMP00X_REGS_CONFIG, reg, 2, 0) < 0) {
+        i2c_release(BUS);
+        return -TMP00X_ERROR_BUS;
+    }
+
+    reg[0] |= (TMP00X_CONFIG_MOD(TMP00X_CONFIG_MOD_CC) >> 8);
+    if (i2c_write_regs(BUS, ADDR, TMP00X_REGS_CONFIG, reg, 2, 0) < 0) {
+        i2c_release(BUS);
+        return -TMP00X_ERROR_BUS;
+    }
+
+    i2c_release(BUS);
+    return TMP00X_OK;
+}
+
+int tmp00x_set_standby(const tmp00x_t *dev)
+{
+    uint8_t reg[2];
+
+    i2c_acquire(BUS);
+    if (i2c_read_regs(BUS, ADDR, TMP00X_REGS_CONFIG, reg, 2, 0) < 0) {
+        i2c_release(BUS);
+        return -TMP00X_ERROR_BUS;
+    }
+
+    reg[0] &= ~(TMP00X_CONFIG_MOD(TMP00X_CONFIG_MOD_CC) >> 8);
+    if (i2c_write_regs(BUS, ADDR, TMP00X_REGS_CONFIG, reg, 2, 0) < 0) {
+        i2c_release(BUS);
+        return -TMP00X_ERROR_BUS;
+    }
+    i2c_release(BUS);
+    return TMP00X_OK;
+}
+
+int tmp00x_read(const tmp00x_t *dev, int16_t *rawv, int16_t *rawt, uint8_t *drdy)
+{
+    uint8_t reg[2];
+
+    i2c_acquire(BUS);
+    /* Register bytes are sent MSB first. */
+    #if defined (MODULE_TMP006)
+    if (i2c_read_regs(BUS, ADDR, TMP00X_REGS_CONFIG, reg, 2, 0) < 0) {
+        i2c_release(BUS);
+        return -TMP00X_ERROR_BUS;
+    }
+    #else
+    if (i2c_read_regs(BUS, ADDR, TMP007_REGS_STATUS, reg, 2, 0) < 0) {
+        i2c_release(BUS);
+        return -TMP00X_ERROR_BUS;
+    }
+    #endif
+    i2c_release(BUS);
+
+    #if defined (MODULE_TMP006)
+    *drdy = reg[1] & (TMP006_CONFIG_DRDY);
+    #else
+    *drdy = reg[0] & (TMP007_CONFIG_DRDY);
+    #endif
+
+    if (!(*drdy)) {
+        LOG_DEBUG("tmp00x_read: conversion in progress!\n");
+        return -TMP00X_ERROR;
+    }
+
+    i2c_acquire(BUS);
+    if (i2c_read_regs(BUS, ADDR, TMP00X_REGS_V_OBJECT, reg, 2, 0) < 0) {
+        i2c_release(BUS);
+        return -TMP00X_ERROR_BUS;
+    }
+    i2c_release(BUS);
+
+    *rawv = ((uint16_t)reg[0] << 8) | reg[1];
+
+    i2c_acquire(BUS);
+    if (i2c_read_regs(BUS, ADDR, TMP00X_REGS_T_AMBIENT, reg, 2, 0) < 0) {
+        i2c_release(BUS);
+        return -TMP00X_ERROR_BUS;
+    }
+    i2c_release(BUS);
+    *rawt = ((uint16_t)reg[0] << 8) | reg[1];
+    return TMP00X_OK;
+}
+
+void tmp00x_convert(int16_t rawv, int16_t rawt,  float *tamb, float *tobj)
+{
+    /* calculate die temperature */
+    *tamb = (double)rawt / 128.0;
```

Comment:
Maybe no magic number and do you really need a `double`?  Can you just use a `float`.  It is a bit more efficient.